EventGenerator/src/PionFilter_module.cc

```diff
@@ -51,26 +57,31 @@ namespace mu2e {
     , diagLevel_{config().diagLevel()}
     , tmin_{config().tmin()}
     , tmax_{config().tmax()}
+    , processCode_{config().processCode()}
     , isNull_{config().isNull()}
   {
   }
-  void PionFilter::beginJob(){}
 
-  bool PionFilter::filter(art::Event& evt) {
+  void PionFilter::beginJob(){
+      art::ServiceHandle<art::TFileService> tfs;
 
+}
+
+  bool PionFilter::filter(art::Event& evt) {
       if(isNull_) return true;
       bool passed = false;
       std::vector<art::Handle<SimParticleCollection>> vah = evt.getMany<SimParticleCollection>();
       for (auto const& ah : vah) { //always one collection
         for(const auto& aParticle : *ah){
           art::Ptr<SimParticle> pp(ah, aParticle.first.asUint());
+
           float _endglobaltime = pp->endGlobalTime();
-          if( pp->stoppingCode() == ProcessCode::hBertiniCaptureAtRest and std::abs(pp->pdgId()) == PDGCode::pi_plus){
+          if( pp->stoppingCode() == processCode_ and std::abs(pp->pdgId()) == PDGCode::pi_plus){
             const PhysicsParams& gc = *GlobalConstantsHandle<PhysicsParams>();
             _totalweight += exp(-1*pp->endProperTime() / gc.getParticleLifetime(pp->pdgId()));
             _ntot += 1;
           }
-          if( pp->stoppingCode() == ProcessCode::hBertiniCaptureAtRest and (std::abs(pp->pdgId()) == PDGCode::pi_plus and _endglobaltime > tmin_ and _endglobaltime < tmax_ )){
+          if( pp->stoppingCode() == processCode_ and (std::abs(pp->pdgId()) == PDGCode::pi_plus and _endglobaltime > tmin_ and _endglobaltime < tmax_ )){
```

Comment:
Not sure why the parens group the tests like this. no parens are needed when all tests are 'and'